src/test/java/com/scalyr/integrations/kafka/AddEventsClientTest.java

```diff
@@ -437,6 +449,52 @@ public void testTooLargeAddEventsSkipped() throws Exception {
     verifyHeaders(request.getHeaders());
   }
 
+  /**
+   * Verify Add Events Requests that exceed maximum add events payload size before compression are not sent.
+   *
+   * In this scenario we test a request which is larger than 6 MB before compression, but smaller than 6 MB
+   * after compression.
+   */
+  @Test
+  public void testTooLargeAddEventsSkippedDeflateCompressedRequest() throws Exception {
+    final int numEvents = 6;
+    final byte[] largeMsgBytes = new byte[1000000];
+    Arrays.fill(largeMsgBytes, (byte)'a');
+    final String largeMsg = new String(largeMsgBytes);
+
+    // Setup Mock Server
+    server.enqueue(new MockResponse().setResponseCode(200).setBody(ADD_EVENTS_RESPONSE_SUCCESS));
+
+    // Create addEvents request
+    AddEventsClient addEventsClient = new AddEventsClient(scalyrUrl, API_KEY_VALUE, ADD_EVENTS_TIMEOUT_MS, ADD_EVENTS_RETRY_DELAY_MS, deflateCompressor);
+    List<Event> events = createTestEvents(numEvents, numServers, numLogFiles, numParsers);
+    events.forEach(event -> event.setMessage(largeMsg));
+
+    assertEquals(0, server.getRequestCount());
+
+    AddEventsResponse addEventsResponse = addEventsClient.log(events).get(5, TimeUnit.SECONDS);;
+    assertEquals("success", addEventsResponse.getStatus());
+    assertEquals("Skipped due to payload too large", addEventsResponse.getMessage());
+
+    // request should be skipped
+    assertEquals(0, server.getRequestCount());
+
+    // Send next batch that is smaller than max payload size
+    events = createTestEvents(numEvents, numServers, numLogFiles, numParsers);
+    addEventsResponse = addEventsClient.log(events).get(5, TimeUnit.SECONDS);;
```

Comment:
Should pass in completable future from previous add events request by calling `AddEventsClient.log(List<Event> events, @Nullable CompletableFuture<AddEventsResponse> dependentAddEvents)`